backend/app/adapter/db/url.go

```diff
@@ -96,6 +97,73 @@ WHERE "%s"=$1;`,
 	return url, nil
 }
 
+// GetByAliases finds URLs for a list of aliases
+func (u URLSql) GetByAliases(aliases []string) ([]entity.URL, error) {
+	// create a string for aliases parameters
+	// it looks like: $1, $2, $3, ......
+	params := make([]string, 0, len(aliases))
+	for i := range aliases {
+		params = append(params, fmt.Sprintf("$%d", i+1))
+	}
+	parameterStr := strings.Join(params, ", ")
+
+	// create a list of interface{} to hold aliases for db.Query()
+	aliasesInterface := []interface{}{}
+	for _, alias := range aliases {
+		aliasesInterface = append(aliasesInterface, alias)
+	}
+
+	var urls []entity.URL
+
+	statement := fmt.Sprintf(`
+SELECT "%s","%s","%s","%s","%s" 
+FROM "%s"
+WHERE "%s" IN (%s);`,
+		table.URL.ColumnAlias,
+		table.URL.ColumnOriginalURL,
+		table.URL.ColumnExpireAt,
+		table.URL.ColumnCreatedAt,
+		table.URL.ColumnUpdatedAt,
+		table.URL.TableName,
+		table.URL.ColumnAlias,
+		parameterStr,
+	)
+
+	stmt, err := u.db.Prepare(statement)
+	if err != nil {
+		return urls, err
+	}
+	defer stmt.Close()
+
+	rows, err := stmt.Query(aliasesInterface...)
```

Comment:
Couldn't find the `for` loop and `QueryRow` in the PR. Please checkout this example mentioned in one of the previous articles:

```go
db, err := sql.Open("postgres", "postgres://postgres:postgres@localhost/postgres?sslmode=disable")
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()
	stmt, err := db.Prepare("select count(*) from pg_stat_activity where datname = $1")
	if err != nil {
		log.Fatal(err)
	}
	defer stmt.Close()
	var num int
	start := time.Now()
	for i := 0; i < 1000; i++ {
		err = stmt.QueryRow("postgres").Scan(&num)
	}
	elapsed := time.Since(start)
	fmt.Printf("got: %d in %s\n", num, elapsed)
```

Reply:
according to official document for function ```QueryRow```:
_**// If the query selects no rows, the *Row's Scan will return ErrNoRows.
// Otherwise, the *Row's Scan scans the first selected row and discards
// the rest.**_

Reply:
it seems that the document is only selecting a single row for testing purpose.
Here we want to select a series of rows, I believe we have to use ```Query```

Reply:
Please read this whole thread: https://github.com/golang/go/issues/5171

Reply:
@byliuyang In that thread, they talk about bulk operations `a way to insert multiple rows at once`.

Our case is a standard SQL query, which requires to call `Query` and pass the list of args.
Please read this http://go-database-sql.org/prepared.html

> Go creates prepared statements for you under the covers. A simple db.Query(sql, param1, param2), for example, works by preparing the sql, then executing it with the parameters and finally closing the statement.

Reply:
@alldroll I want to make everything explicit to avoid magics.
@dawntomm I don't want to block your changes. Let's add a TODO here to further discuss our SQL handling strategy. I'll approve this PR after the TODO is added.

Reply:
@byliuyang Just added a TODO

Reply:
@byliuyang sorry I still did not get why you think we should use QueryRow instead of Query, if we use QueryRow then it will cost us more connections with DB, we could definitely do performance test, but intuitively I feel it's slower

Reply:
@dawntomm All queryRows in a single prepare statement share one connection.